ToolChain/Assembler/Parser/Parser.py

```diff
@@ -585,3 +548,131 @@ def _generateBinaryInstructionFromInstruction(self, instruction: Instruction=Non
                                                                               memoryRefSymbol.encode())
 
         return binaryInstructionByteString
+
+    def translateRegisterNameToRegisterCode(self, registerName: str=""):
+        """
+        This takes a register name and returns a register code as per:
+            A = 0b0000
+            B = 0b0001
+            C = 0b0010
+            ...
+            etc.
+        Throws error if register is not part of valid list within system.
+        :param registerName: str, representing the register that needs translation
+        :return: int, the int that represents the register
+        """
+
+        registerCode = None
+        registerName = registerName.upper()
+
+        if registerName == "A":
+            registerCode = REGISTER_A
+        elif registerName == "B":
+            registerCode = REGISTER_B
+        elif registerName == "C":
+            registerCode = REGISTER_C
+        elif registerName == "D":
+            registerCode = REGISTER_D
+        elif registerName == "E":
+            registerCode = REGISTER_E
+        elif registerName == "F":
+            registerCode = REGISTER_F
+        elif registerName == "G":
+            registerCode = REGISTER_G
+        elif registerName == "A2":
+            registerCode = REGISTER_A2
+        elif registerName == "B2":
+            registerCode = REGISTER_B2
+        elif registerName == "C2":
+            registerCode = REGISTER_C2
+        elif registerName == "D2":
+            registerCode = REGISTER_D2
+        elif registerName == "E2":
+            registerCode = REGISTER_E2
+        elif registerName == "F2":
+            registerCode = REGISTER_F2
+        elif registerName == "G2":
+            registerCode = REGISTER_G2
+        elif registerName == "S":
+            registerCode = REGISTER_S
+        elif registerName == "S2":
+            registerCode = REGISTER_S2
+        else:
+            raise ValueError("Invalid register provided. '{}' seen as input, expect valid register."
+                             .format(registerName))
+
+        return registerCode
+
+    def translateTextImmediateToImmediate(self, textImmediate: str=""):
+        """
+        This will translate an immediate value in a way that can be understood by the architecture.
+        :param textImmediate: str, an immediate value to be translated
+        :return: int, an immediate that can be worked on
+        """
+
+        immediate = None
+        isNegative = False
+        textImmediate = textImmediate.lower()  # Needed in case of 0XFF instead of 0xFF
+
+        if textImmediate[0] == "-":
```

Comment:
Maybe we want to add a new constant to 
spartacus/ToolChain/Assembler/Constants.py